src/core/metallicroughnessmaterial.cpp

```diff
@@ -971,6 +999,14 @@ void MetallicRoughnessMaterial::setToneMappingAlgorithm(MetallicRoughnessEffect:
     m_effect->setToneMappingAlgorithm(algorithm);
 }
 
+void MetallicRoughnessMaterial::setMorphController(MorphController *morphController)
+{
+    if (morphController == m_morphControllerParameter->value().value<MorphController *>())
+        return;
+
+    m_morphControllerParameter->setValue(QVariant::fromValue(morphController));
```

Comment:
Im not sure I get this. You are setting a pointer to a custom class for the parameter, which is the uniform in the shader?

Reply:
MorphController is a ShaderData, setting a ShaderData on a QParameter is possible and is used when you need to set a struct for a uniform